core/src/metrics/registry.rs

```diff
@@ -217,6 +219,28 @@ impl MetricsRegistryTrait for MetricsRegistry {
         }
     }
 
+    fn global_histogram_vec(
+        &self,
+        name: &str,
+        help: &str,
+        variable_labels: &[&str],
+    ) -> Result<HistogramVec, PrometheusError> {
+        let opts = HistogramOpts::new(name, help);
+        let histograms = HistogramVec::new(opts, variable_labels)?;
+        let id = histograms.desc().first().unwrap().id;
+        let maybe_histogram = self.global_histogram_vecs.read().unwrap().get(&id).cloned();
+        if let Some(histograms) = maybe_histogram {
+            Ok(histograms)
+        } else {
+            self.register(name, Box::new(histograms.clone()));
+            self.global_histogram_vecs
+                .write()
+                .unwrap()
```

Comment:
Just to confirm, all of these `.unwrap`s are safe to do right?

Reply:
This by definition only panics if something else has already panicked, and in this use there is no possibility of panic while holding the lock.